source/parser.hera

```diff
@@ -6859,13 +6870,15 @@ Init
 
     // Gather nodes that match a predicate recursing into all unmatched children
     // i.e. if the predicate matches a node it is not recursed into further
-    function gatherRecursive(node, predicate) {
+    function gatherRecursive(node, predicate, skipPredicate) {
       if (node == null) return []
 
       if (Array.isArray(node)) {
         return node.flatMap((n) => gatherRecursive(n, predicate))
       }
 
+      if (skipPredicate?.(node)) return
```

Comment:
We might want to `return []` here so that:

```
gatherRecursive(s, ->, -> true).forEach(...)
```

will work. Also it won't insert `undefined` into the flatmap.